mlir/include/mlir/Dialect/EmitC/IR/EmitC.td

```diff
@@ -835,6 +835,25 @@ def EmitC_LogicalOrOp : EmitC_BinaryOp<"logical_or", [CExpression]> {
   let assemblyFormat = "operands attr-dict `:` type(operands)";
 }
 
+def EmitC_LValueLoadOp : EmitC_Op<"lvalue_load", [
```

Comment:
Can we shorten to `LoadOp` and `emitc.load`? Encoding the required type in the name doesn't feel right.
Would also match `AssignOp` and `emitc.assign` which now also require an lvalue.